src/awkward/operations/ak_to_parquet.py

```diff
@@ -196,16 +252,28 @@ def to_parquet(
         primitive_policy="error",
         string_policy="as-characters",
     )
```

Comment:
```python
layout = ak.operations.ak_to_layout._impl(data, ...)
```

will convert `data` into an Awkward Array, and Awkward Arrays are eagerly in memory all at once. `ak.to_parquet_row_groups` is supposed to take an iterator as `data`, so this will iterate over it, pulling it all into memory, which is not what you want.

In the case of `ak.to_parquet_row_groups`, you'll want to do a Python for loop over `data` and call `ak_to_layout` on each item individually.

Reply:
Okay! Now calling to_layout for each iteration - I printed each new layout to check that they only have data from one array and it looks good, but I'll test it on a large dataset to be sure

Reply:
I made a test and was able to write file larger than my laptop's memory!

<img width="398" alt="Screenshot 2024-01-29 at 11 54 11" src="https://github.com/scikit-hep/awkward/assets/70441641/2229d316-4c67-4da7-9a28-55fc425cd1b2">

Reply:
Also, the output file is compressed[^1], whereas the data in memory were not compressed.